### PR TITLE
Supported gnu auto generated test cases

### PR DESCRIPTION
**Pull request by yulong-plct**
Hi, @eopXD :
This PR adds the implementation for gnu auto generated test case that I did together with @Incarnation-p-lee and @xl0714.
Please review it.
It is worth noting that this PR only generates cases, and the rest of the content will be in another PR.
Thanks!

### CODE REVIEW

**Review by eopXD — COMMENTED** [review]
Thank you for the organized PR!

- Nit: In my opinion, the commit "create folder for gnu auto generated test cases" should be ordered after your edits to the generator. Since `--toolchain-type` is not yet supported at the first place.
- Please capitalize the first letter of your commit message
- Please add another commit to separate the current target in the makefile `gen-test` to consist of `gen-llvm-test` and `gen-gnu-test`. Then, please try to commit the test case additions under `auto-generated/`  with `make git-commit-all` and see if things work out.
- Please check the CI under your branch, make sure the CI passes.

File: rvv-intrinsic-generator/Makefile [hunk 1]
```diff
@@ -106,9 +106,9 @@ define gen_docs
 endef
 
 define gen_tests
-	rm -f $(1)/*.c
+	rm $(1)/*.c $(1)/*.C -f
```
Comment (Incarnation-p-lee) [inline rvv-intrinsic-generator/Makefile:109]: I suppose we don't generate source files like "*.C" anymore, @eopXD  can help to confirm about this.

File: rvv-intrinsic-generator/Makefile [hunk 2]
```diff
@@ -137,6 +137,7 @@ endif
 gen-document: non-overloaded-doc non-overloaded-docs overloaded-doc overloaded-docs
 gen-test: non-overloaded-test overloaded-test llvm-non-overloaded-test llvm-overloaded-test
 gen-compatible-header: non-policy-compatible-header policy-compatible-header non-policy-overloaded-compatible-header policy-overloaded-compatible-header
+gen-gnu-test: gnu-overloaded-test gnu-non-overloaded-test
```
Comment (Incarnation-p-lee) [inline rvv-intrinsic-generator/Makefile:140]: who will consume this target `gen-gnu-test` when overall making?

File: rvv-intrinsic-generator/Makefile [hunk 3]
```diff
@@ -170,18 +171,28 @@ overloaded-test:
...
-	$(call gen_tests,$(DIR)/llvm-overloaded-tests,overloaded-test,--llvm)
-	$(call gen_tests,$(POLICY_DIR)/llvm-overloaded-tests,overloaded-test,--llvm --has-policy)
+	$(call gen_tests,$(DIR)/llvm-overloaded-tests,overloaded-test,--toolchain-type llvm)
+	$(call gen_tests,$(POLICY_DIR)/llvm-overloaded-tests,overloaded-test,--toolchain-type llvm --has-policy)
 	$(call replace_float, $(DIR)/llvm-overloaded-tests)
 	$(call replace_float, $(POLICY_DIR)/llvm-overloaded-tests)
 
+# Generate GNU non-overloaded intrinsic testing source files
+gnu-non-overloaded-test:
+	$(call gen_tests,$(DIR)/gnu-api-tests,non-overloaded-test,--toolchain-type gnu)
```
Comment (Incarnation-p-lee) [inline rvv-intrinsic-generator/Makefile:188]: I think the first PR would be adding the gnu target without any further action to take, and then committing the gnu auto-generated folder.

You can reference [this](http://principles-wiki.net/principles:keep_it_simple_stupid) and [this](https://arxiv.org/vc/cs/papers/0701/0701021v1.pdf) doc for more details.

**Review by Incarnation-p-lee — CHANGES_REQUESTED** [review]

**Comment by yulong-plct** [comment]
I will make modifications according to the above suggestions and submit a new PR.
Thanks a lot! @eopXD @Incarnation-p-lee

**Comment by eopXD** [comment]
> I will make modifications according to the above suggestions and submit a new PR. Thanks a lot! @eopXD @Incarnation-p-lee

I think you don't need to open up another PR, you can force push to the same brunch and this PR will be updated.

**Comment by eopXD** [comment]
Additionally, I think we can also add a Github Action to verify the test cases you have generated here upon the latest GNU toolchain here, or have you already planned to include this in a separate PR?

**Pull request closed without merging**